src/hooks/useIsValidExecution.ts

```diff
@@ -20,14 +51,38 @@ const useIsValidExecution = (
   executionValidationError?: Error
   isValidExecutionLoading: boolean
 } => {
-  const safeSdk = useSafeSDK()
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+  const readOnlyProvider = useWeb3ReadOnly()
+  const isOwner = useIsSafeOwner()
 
   const [isValidExecution, executionValidationError, isValidExecutionLoading] = useAsync(async () => {
-    if (!safeTx || !safeSdk || !gasLimit) {
+    if (!safeTx || !wallet || !gasLimit || !readOnlyProvider) {
       return
     }
+
     try {
-      return await safeSdk.isValidTransaction(safeTx, { gasLimit: gasLimit.toString() })
+      const provider = getPatchedSignerProvider(wallet, safe.chainId, readOnlyProvider)
+      const { contract } = getSpecificGnosisSafeContractInstance(safe, provider)
+
+      /**
+       * We need to call the contract directly instead of using sdk.isValidTransaction
+       * because gasLimit errors are not propagated otherwise. It also fixes the over-fetching
+       * issue that the monkey patched provider does
+       */
+      return contract.callStatic.execTransaction(
```

Comment:
I believe the root cause is that we don't throw an error / swallow errors in the core-sdk: https://github.com/safe-global/safe-core-sdk/blob/main/packages/safe-ethers-lib/src/contracts/GnosisSafe/GnosisSafeContractEthers.ts#L126

We could fix it there and use the sdk call again I think.